packages/shared/lib/services/proxy/request.ts

```diff
@@ -3,72 +3,88 @@ import type { AxiosResponse, AxiosRequestConfig } from 'axios';
 import type { Result, RetryAttemptArgument } from '@nangohq/utils';
 import { axiosInstance as axios, Err, Ok, redactHeaders, redactURL, retryFlexible } from '@nangohq/utils';
 
-import type { ApplicationConstructedProxyConfiguration, MaybePromise, MessageRowInsert } from '@nangohq/types';
-import { buildProxyHeaders, buildProxyURL, ProxyError } from './utils.js';
+import type { ApplicationConstructedProxyConfiguration, ConnectionForProxy, MaybePromise, MessageRowInsert } from '@nangohq/types';
+import { getAxiosConfiguration, ProxyError } from './utils.js';
 import { getProxyRetryFromErr } from './retry.js';
 
+interface Props {
+    proxyConfig: ApplicationConstructedProxyConfiguration;
+    logger: (msg: MessageRowInsert) => MaybePromise<void>;
+    onError?: (args: { err: unknown; max: number; attempt: number }) => { retry: boolean; reason: string; wait?: number };
+    getConnection: () => MaybePromise<ConnectionForProxy>;
+}
+
```

Comment:
Rewrote this file to accept a getConnection param that's called on every iteration